File: android/FluppyChat/app/src/main/java/eggs/painted/fluppychat/Util/Decoder.java
package eggs.painted.fluppychat.Util;

import android.util.Log;

import com.scottyab.aescrypt.AESCrypt;

import java.security.GeneralSecurityException;

/**
 * Created by dmytro on 27.07.15.
 */
final public class Decoder {
    static private String PASSWORD = "password";

    /**
     * Code text
     *
     * @param text - message for coding
     * @return - coded message
     */
    static public String codeMessage( final String text ) {
        String encryptedMsg = null;

        try {
            encryptedMsg = AESCrypt.encrypt( PASSWORD, text );
        }catch (GeneralSecurityException e){
            e.printStackTrace();
            Log.e("ECNCRYPT_ERROR", "error ocqurence when encrypt message");
            //handle error
        }

        return encryptedMsg;
    }

    /**
     * Decode text
     *
     * @param text - message for decoding
     * @return - decoded message
     */
    static public String decodeMessage( final String text ) {
        String msg = null;

        try {
            msg = AESCrypt.decrypt(PASSWORD, text);
        } catch (GeneralSecurityException e){
            e.printStackTrace();
            Log.e("DECODE_MESSAGE", "Error when decode message");
        } catch ( Exception e ){
            e.printStackTrace();
            Log.e( "DECODE_MESSAGE", "Error when decode message" );
        }

        return msg;
    }
}

File: ios/FluppyChat/FluppyChat/Util/Decoder.swift
//
//  Decoder.swift
//  FluppyChat
//
//  Created by Dmytro Bohachevskyy on 10/1/15.
//  Copyright © 2015 Dmytro Bohachevskyy. All rights reserved.
//

class Decoder {
    let PASSWORD = "password"
    
    func decode(text : String) -> String {
        return AESCrypt.decrypt(text, password: PASSWORD)
    }
    
    func encode(text : String) -> String {
        return AESCrypt.encrypt(text, password: PASSWORD)
    }

}
